lib/librte_hash/rte_cuckoo_hash.c

```diff
@@ -1339,6 +1339,215 @@ rte_hash_lookup_with_hash(const struct rte_hash *h,
 	return __rte_hash_lookup_with_hash(h, key, sig, NULL);
 }
 
+/* Search one bucket to find the match key - uses rw lock and yield */
+static inline int32_t
+search_one_bucket_and_yield_l(const struct rte_hash *h, const void *key,
+	uint16_t sig, void **data, const struct rte_hash_bucket *bkt,
+	rte_hash_yield_func yield_func, void *arg)
+{
+	int i;
+	struct rte_hash_key *k, *keys = h->key_store;
+
+	for (i = 0; i < RTE_HASH_BUCKET_ENTRIES; i++) {
+		if (bkt->sig_current[i] == sig &&
+				bkt->key_idx[i] != EMPTY_SLOT) {
+			k = (struct rte_hash_key *) ((char *)keys +
+					bkt->key_idx[i] * h->key_entry_size);
+			yield_func(k, arg);
```

Comment:
In this function call, we are yielding while the lock is acquired.

Reply:
It is **now** okay because we are yielding with a reader lock.